examples/EDMF_data/uq_for_edmf.jl

```diff
@@ -0,0 +1,246 @@
+            exp_dir *
+            "/\" not found. Please unzip the compressed file \"" *
+            exp_name *
+            ".zip\" and retry.",
+        )
+    end
+    data_filepath = joinpath(exp_dir, "Diagnostics.nc")
+    if !isfile(data_filepath)
+        LoadError("experiment data file \"Diagnostics.nc\" not found in directory \"" * exp_dir * "/\"")
+    else
+        y_truth = Array(NCDataset(data_filepath).group["reference"]["y_full"]) #ndata
+        truth_cov = Array(NCDataset(data_filepath).group["reference"]["Gamma_full"]) #ndata x ndata
+        # Option (i) get data from NCDataset else get from jld2 files.
+        output_mat = Array(NCDataset(data_filepath).group["particle_diags"]["g_full"]) #nens x ndata x nit
+        input_mat = Array(NCDataset(data_filepath).group["particle_diags"]["u"]) #nens x nparam x nit
+        input_constrained_mat = Array(NCDataset(data_filepath).group["particle_diags"]["phi"]) #nens x nparam x nit
+
+        #TODO: Remove mask after testing
+        mask = 1:size(y_truth, 1)
+        y_truth = y_truth[mask]
+        truth_cov = truth_cov[mask, mask]
+        output_mat = output_mat[:, mask, :]
+
+        # to be consistent, we wish to go from nens x nparam x nit arrays to nparam x (nit x nens) 
+        inputs =
+            reshape(permutedims(input_mat, (2, 3, 1)), (size(input_mat, 2), size(input_mat, 3) * size(input_mat, 1)))
+        inputs_constrained = reshape(
+            permutedims(input_constrained_mat, (2, 3, 1)),
+            (size(input_constrained_mat, 2), size(input_constrained_mat, 3) * size(input_constrained_mat, 1)),
+        )
+
+        outputs = reshape(
+            permutedims(output_mat, (2, 3, 1)),
+            (size(output_mat, 2), size(output_mat, 3) * size(output_mat, 1)),
+        )
+
+        # naive preprocessing of data
+        #remove extremes using truth cov
+        iqr_rows = mapslices(iqr, outputs, dims = 2)
+        med = mapslices(median, outputs, dims = 2)
+        twentyfive_rows = mapslices(x -> percentile(x, 25), outputs, dims = 2)
+        seventyfive_rows = mapslices(x -> percentile(x, 75), outputs, dims = 2)
+        max_bd = seventyfive_rows + 5 * iqr_rows
+        min_bd = twentyfive_rows - 5 * iqr_rows
+
+        idx_in_bd = (outputs .- min_bd) .> 0 .&& (max_bd .- outputs) .> 0
+        #remove all-bad particles
+        not_all_bad_particles = vec(any(idx_in_bd, dims = 1))
+        outputs = outputs[:, not_all_bad_particles]
+        inputs = inputs[:, not_all_bad_particles]
+        inputs_constrained = inputs_constrained[:, not_all_bad_particles]
+        println("Pruning all-bad particles...")
+        println("removed ", length(not_all_bad_particles)[1] - sum(not_all_bad_particles), " particles.")
+
+        #remove all-bad data dimensions
+        idx_in_bd = (outputs .- min_bd) .> 0 .&& (max_bd .- outputs) .> 0
+        not_all_bad_data_points = vec(any(idx_in_bd, dims = 2))
+        outputs = outputs[not_all_bad_data_points, :]
```

Comment:
why do we need this step, if you are already filtering outlying particles in L114-L119?

should I understand this as a cropping of dimensions that have structural or model errors outside of the range of the uncertainty given by the truth covariance?